webapp/src/components/setup_ui/setup_ui.jsx

```diff
@@ -11,8 +11,8 @@ import JiraIcon from 'components/icon';
 // SetupUI is a dummy Root component that we use to detect when the user has logged in
 export default class SetupUI extends PureComponent {
     static propTypes = {
-        userConnected: PropTypes.bool.isRequired,
-        instanceInstalled: PropTypes.bool.isRequired,
+        userConnected: PropTypes.bool,
+        instanceInstalled: PropTypes.bool,
```

Comment:
If these are booleans, can we not pass in false instead of `null` or `undefined` while we are not connected/installed?

Reply:
@mickmister They are coming from the redux store for the plugin, and are set to false as defaults. But what seems to be happening is the the SetupUI component is mounted before the redux store (for the plugin) has a chance to initialize. I figure it's not that big of a deal at the moment. Being undefined works just as well for a default. 2/5 I don't want to manually override the redux state in the index file -- that might cause problems later when you least expect it...

Reply:
> The SetupUI component is mounted before the redux store (for the plugin) has a chance to initialize.

Do you know why this is? That seems like a bug to me. In a normal react app, the original reducer values are available upon first render. What is different about how we set up the redux store here?

I'm sorry about my lack of knowledge here. I need to study more how the webapp plugins are bootstrapped.

Reply:
@mickmister I think I know why now -- it's because we haven't initialized the reducers until we register them in the setupUILater function. That's why they haven't been defaulted yet. Makes sense now.